test/Sentry.Tests/SentrySdkTests.cs

```diff
@@ -95,17 +95,19 @@ public void Init_CallbackWithoutDsn_ValidDsnEnvironmentVariable_LocatesDsnEnviro
         }
 
         [Fact]
-        public void Init_CallbackWithoutDsn_InvalidDsnEnvironmentVariable_DisabledSdk()
```

Comment:
No idea why this expected it to return `DisabledSdk` considering that a similar test expects an exception: `Init_InvalidDsnEnvironmentVariable_Throws`. I believe throwing an exception is the way to go here.

Reply:
Invalid DSN is a clear mistake so makes sense to throw.

I wouldn't disagree that if `Dsn` is `null` we can throw (only "" is disable) at init but this is a breaking change

Reply:
I think the key point here is that (in the current state) when the user sets `SENTRY_DSN` environment variable to an invalid dsn, it will just be ignored. I think if that environment variable is set, it was a conscious decision, no less than setting it in `appsettings.json`, so it should throw an exception instead.

Reply:
Makes sense.